src/undo-plugin.ts

```diff
@@ -177,38 +177,26 @@ export function canRedo(state: EditorState): boolean {
   return undoState?.undoManager.canRedo() || false;
 }
 
-export function undo(
-  state: EditorState,
-  dispatch: (tr: Transaction) => void,
-): boolean {
+export const undo: Command = (state, dispatch): boolean => {
   const undoState = loroUndoPluginKey.getState(state);
-  if (!undoState || !undoState.undoManager.canUndo()) {
+  if (!undoState) {
     return false;
   }
-
-  if (!undoState.undoManager.undo()) {
-    const emptyTr = state.tr;
-    dispatch(emptyTr);
-    return false;
```

Comment:
There is no point calling `dispatch` when the function failed to execute the action (i.e. return `false`).